dask/array/linalg.py

```diff
@@ -822,6 +822,8 @@ def norm(x, ord=None, axis=None, keepdims=False):
             raise ValueError("Invalid norm order for vectors.")
 
         r = svd(x)[1].sum()
+        if keepdims:
+            r = r[None, None]
```

Comment:
Maybe we should just pass `keepdims=keepdims` to `sum`.

Reply:
I think that only prevents 1 dimension from being removed. We need the output to be 2 dimensions.

Reply:
Although maybe all of these could be replaced by a single check just before returning `r`?

Reply:
Shouldn't. That would be a bug.

With NumPy...

```python
In [1]: np.arange(6).reshape(2, 3).sum(keepdims=True)
Out[1]: array([[15]])
```

Reply:
Oh, I think it's the slice that removes a dimension, and then the reduction that removes the second.

Reply:
So `svd` returns multiple values. The index is just selecting which one to use.

ref: https://docs.scipy.org/doc/numpy/reference/generated/numpy.linalg.svd.html

Reply:
I think I see what you mean now. The middle argument is a vector. So we need to add another dimension to it in order to get a matrix back. It's a strange case to have `keepdims` in, but it make sense.

Reply:
Adding one singleton dimension before the reduction function with these `svd` cases and then calling the reduction function with `keepdims` did the trick for me.